Lib/test/test_sax.py

```diff
@@ -774,8 +774,8 @@ class StreamReaderWriterXmlgenTest(XmlgenTest, unittest.TestCase):
     fname = support.TESTFN + '-codecs'
 
     def ioclass(self):
-        writer = codecs.open(self.fname, 'w', encoding='ascii',
-                             errors='xmlcharrefreplace', buffering=0)
+        writer = open(self.fname, 'w', encoding='ascii',
```

Comment:
The code of `XMLGenerator` is complex, it contains a lot of special cases and should be tested for different kinds of writers.

It would be better to add a test for `io.TextIOWrapper` (if testing `io.StringIO` is not enough), but keep the test for `StreamReaderWriter`.